src/rate_limiter.rs

```diff
@@ -1,52 +1,93 @@
+use crate::schema::{publish_limit_buckets, publish_rate_overrides};
+use crate::sql::{date_part, floor, greatest, interval_part, least, pg_enum};
+use crate::util::errors::{AppResult, TooManyRequests};
 use chrono::{NaiveDateTime, Utc};
-use diesel::data_types::PgInterval;
+use diesel::dsl::IntervalDsl;
 use diesel::prelude::*;
 use diesel::sql_types::Interval;
+use std::borrow::Cow;
+use std::collections::HashMap;
 use std::time::Duration;
 
-use crate::schema::{publish_limit_buckets, publish_rate_overrides};
-use crate::sql::{date_part, floor, greatest, interval_part, least, pg_enum};
-use crate::util::errors::{AppResult, TooManyRequests};
-
 pg_enum! {
     pub enum LimitedAction {
         PublishNew = 0,
+        PublishUpdate = 1,
+        YankUnyank = 2,
+    }
+}
+
+impl LimitedAction {
+    pub fn default_rate_seconds(&self) -> u64 {
+        match self {
+            LimitedAction::PublishNew => 60 * 60,
+            LimitedAction::PublishUpdate => 60,
+            LimitedAction::YankUnyank => 60,
+        }
+    }
+
+    pub fn default_burst(&self) -> i32 {
+        match self {
+            LimitedAction::PublishNew => 5,
+            LimitedAction::PublishUpdate => 30,
+            LimitedAction::YankUnyank => 30,
```

Comment:
we talked about this PR in the team meeting on Friday and felt that the burst limit for yanks might be a little on the low side. could you change this to 100? :)

```suggestion
            LimitedAction::YankUnyank => 100,
```

Reply:
Will update it.